### PR TITLE
Task/tec 262/location autocomplete

### PR DESCRIPTION
**Pull request by Benrobo**
- feat: Added location suggestions based on `country` selected only.
- Added retrieval of location coordinates when `updating` or `registering` a new businesses

> MapBox doesn't support searching location based on `state` and `city` but offers filtering location by `states` and `city.` Same with Google Geolocation API.

### CODE REVIEW

**Comment by osxar** [comment]
I noticed that the error returned was too descriptive and exposed a lot of personal data. Let's not return the true error like this because of security reason. Instead return an descriptive error message were possible, and we can `console.error` log the full error in the backend for debugging purposes.

Screenshot 2024-09-27 at 12 09 02 AM

**Comment by osxar** [comment]
> * feat: Added location suggestions based on `country` selected only.
> * Added retrieval of location coordinates when `updating` or `registering` a new businesses
> 
> > MapBox doesn't support searching location based on `state` and `city` but offers filtering location by `states` and `city.` Same with Google Geolocation API.

Let's include the selected `state` and `city` in the filtering options for Canada, while for Nigeria, we can use `state` filtering option only. Let's connect tomorrow and i can explain why

**Comment by Benrobo** [comment]
> I noticed that the error returned was too descriptive and exposed a lot of personal data. Let's not return the true error like this because of security reason. Instead return an descriptive error message were possible, and we can `console.error` log the full error in the backend for debugging purposes.
> 
> Screenshot 2024-09-27 at 12 09 02 AM

This error message resulted from the pattern and approach used when handling errors in the backend, I requested to update this initially when I joined, but unfortunately it wasn't granted.

though I have a better way of handling this on the frontend, but since the error Object message is been sent as a "description" from backend, nothing can be done here:

```js
export const getAxiosErrorMessage = (
  error: AxiosError & {
    response?: { data?: { message?: string | { desc: string } } };
  },
  message?: string
): string => {
  if (typeof error?.response?.data?.message === "string") {
    return error.response.data.message;
  }
  if (typeof error?.response?.data?.message?.desc === "string") {
    return error.response.data.message.desc;
  }
  return error?.message || message || "An unknown error occurred";
};
```

File: src/app/register-business/page.tsx [hunk 1]
```diff
@@ -154,8 +159,16 @@ function RegisterBusinessPage() {
     },
   });
   const updateBusinessProfileMut = useMutation({
-    mutationFn: async (props: { data: UpdateBusinessBody; id: string }) =>
-      await updateUserBusinessProfileDetail(props.id, props.data),
+    mutationFn: async (props: {
+      data: UpdateBusinessBody;
+      id: string;
+      session_token: string;
```
Comment (osxar) [inline src/app/register-business/page.tsx:165]: @Benrobo since we aren't using session, please remove all the session implemetation

**Review by osxar — COMMENTED** [review]
Create another PR and remove all the session implementation

**Review by osxar — APPROVED** [review]

**Pull request merged**